src/Feature/Syntax/Feature_Syntax.re

```diff
@@ -95,3 +95,23 @@ let handleUpdate = (bufferUpdate: BufferUpdate.t, highlights: t) => {
     highlights,
   );
 };
+
+let update = (highlights: t, msg) =>
+  switch (msg) {
+  | TokensHighlighted(tokens) => setTokens(tokens, highlights)
+  | BufferUpdated(bu) when !bu.isFull => handleUpdate(bu, highlights)
+  | _ => highlights
```

Comment:
When you have a well-defined set of variants, you should enumerate all of them. Exhaustiveness checking is a very useful tool that allows you to have a "conversation" with the compiler when adding a new variant as part of adding functionality. I therefore usually consider the wildcard pattern a code smell.

Reply:
Good point. I implemented the exhaustive check and took out the wildcard.